src/main/java/kroppeb/stareval/parser/TokenStack.java

```diff
@@ -0,0 +1,150 @@
+package kroppeb.stareval.parser;
+
+import kroppeb.stareval.token.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+class TokenStack {
+	final List<Token> stack = new ArrayList<>();
```

Comment:
Why not use a `Deque<Token>`? So that it works like an actual stack instead of doing stuff with a `List`

Reply:
An `ArrayList` is a valid stack implementation. A `Dequeue` is a queue and stack in one. I usually prefer to use lists when I just need a stack, but if you think it's better to switch to a dequeue than I could switch.

Reply:
The main difference I can see between a `Deque` and `List` when implemented as a stack is that `Deque`s iterate in a reverse order compared to `List`s (which makes more sense, considering the nature of a stack having the top element be accessed first)
Generally a `Deque` is the preferred stack implementation (in my experience), with native `push()` and `pop()` and `peek()` methods as well as in many cases being faster (likely why it is used in things like vanilla's own MatrixStack), but whatever you think works best is alright. You may resolve this conversation if you see fit.